src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -281,13 +385,26 @@ def q_impl(
             assert numpy.isclose(q_bias.quantizer.scale, m_matmul)
             numpy_q_out += q_bias.qvalues
 
-        with tag(self.op_instance_name + ".matmul_rounding"):
-            # Apply Concrete rounding (if relevant)
-            numpy_q_out = self.cnp_round(numpy_q_out, calibrate_rounding)
-
-        # Quantization scales and zero points (FLOATS involved)
-        # This is going to be compiled with a PBS (along with the following activation function)
-        numpy_q_out = numpy_q_out.astype(numpy.float64) + final_term - sum_weights
+        # If weights are not encrypted then we can round as the next
+        # line is going ot be done in a PBS
+        if not is_encrypted_gemm:
+            with tag(self.op_instance_name + ".matmul_rounding"):
+                # Apply Concrete rounding (if relevant)
+                numpy_q_out = self.cnp_round(numpy_q_out, calibrate_rounding)
+
+                # Force a PBS with astype float64
+                numpy_q_out = numpy_q_out.astype(numpy.float64)
+
+        # Quantization scales and zero points
+        # This is done in a PBS if is_encrypted_gemm == False
+        # Otherwise it is done in clear and be compiled with a PBS
```

Comment:
>        # Otherwise it is done in clear and be compiled with a PBS
        # (along with the following activation function)

done in the clear and "be compiled with a PBS" ? I think there's something wrong in this comment, or I don't get it

Reply:
Indeed very unclear. I meant, "Otherwise it is done in FHE". The end must be something I forgot to remove. Thanks!